[Bug]: `DeprecationWarning: pkg_resources is deprecated as an API` on `=pynwb-2.3.1`

### What happened?

The latest pynwb gives a deprecation warning, which in DANDI is elevated to an error, making all tests which rely on pynwb fail.

### Steps to Reproduce

```python
(dev) [deco]~/src/dandi-cli ❱ pytest -vvs dandi/cli/tests/test_cmd_validate.py::test_process_issues
ImportError while loading conftest '/home/chymera/src/dandi-cli/dandi/conftest.py'.
dandi/conftest.py:7: in <module>
    from .tests.fixtures import *  # noqa: F401, F403  # lgtm [py/polluting-import]
dandi/tests/fixtures.py:20: in <module>
    import pynwb
/usr/lib/python3.10/site-packages/pynwb/__init__.py:105: in <module>
    __resources = __get_resources()
/usr/lib/python3.10/site-packages/pynwb/__init__.py:25: in __get_resources
    from pkg_resources import resource_filename
/usr/lib/python3.10/site-packages/pkg_resources/__init__.py:121: in <module>
    warnings.warn("pkg_resources is deprecated as an API", DeprecationWarning)
E   DeprecationWarning: pkg_resources is deprecated as an API
(dev) [deco]~/src/dandi-cli ❱ eix pynwb
[?] dev-python/pynwb [1]
     Available versions:  (~)2.1.0-r2^t (~)2.2.0-r1^t {test PYTHON_TARGETS="python3_10"}
     Installed versions:  2.3.1(15:37:18 22/03/23)(test PYTHON_TARGETS="python3_10")
     Homepage:            https://github.com/NeurodataWithoutBorders/pynwb
     Description:         A Python API for working with Neurodata stored in the NWB Format

[1] "science" /home/chymera/src/sci
(dev) [deco]~/src/dandi-cli ❱ python
Python 3.10.10 (main, Mar 20 2023, 10:28:47) [GCC 12.2.1 20230304] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pynwb
>>> pynwb.__version__
'2.3.1'
```


### Traceback

_No response_

### Operating System

Linux

### Python Executable

Python

### Python Version

3.10

### Package Versions

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you checked the [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this bug was not already [reported](https://github.com/NeurodataWithoutBorders/pynwb/issues)?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@
 ### Bug fixes
 - Fixed bug in ``NWBHDF5IO.nwb_version`` property to support files written by third-party software with a fixed-length ``nwb_version`` attribute. @oruebel [#1669](https://github.com/NeurodataWithoutBorders/pynwb/pull/1669)
 - Fixed search bar and missing jquery in ReadTheDocs documentation. @rly [#1671](https://github.com/NeurodataWithoutBorders/pynwb/pull/1671)
+- Fixed issue with deprecated pkg_resources. @rly [#1678](https://github.com/NeurodataWithoutBorders/pynwb/pull/1678)
 
 ## PyNWB 2.3.1 (February 24, 2023)
 

--- a/src/pynwb/__init__.py
+++ b/src/pynwb/__init__.py
@@ -15,16 +15,24 @@ from hdmf.build import BuildManager, TypeMap
 import hdmf.common
 
 CORE_NAMESPACE = 'core'
-__core_ns_file_name = 'nwb.namespace.yaml'
 
 from .spec import NWBDatasetSpec, NWBGroupSpec, NWBNamespace  # noqa E402
 from .validate import validate  # noqa: F401, E402
 
 
 def __get_resources():
-    from pkg_resources import resource_filename
+    try:
+        from importlib.resources import files
+    except ImportError:
+        # TODO: Remove when python 3.9 becomes the new minimum
+        from importlib_resources import files
+
+    __location_of_this_file = files(__name__)
+    __core_ns_file_name = 'nwb.namespace.yaml'
+    __schema_dir = 'nwb-schema/core'
+
     ret = dict()
-    ret['namespace_path'] = os.path.join(resource_filename(__name__, 'nwb-schema/core'), __core_ns_file_name)
+    ret['namespace_path'] = str(__location_of_this_file / __schema_dir / __core_ns_file_name)
     return ret
 
 

--- a/tests/back_compat/test_import_structure.py
+++ b/tests/back_compat/test_import_structure.py
@@ -34,7 +34,6 @@ class TestImportStructure(TestCase):
             "__TYPE_MAP",
             "__builtins__",
             "__cached__",
-            "__core_ns_file_name",
             "__doc__",
             "__file__",
             "__get_resources",

